servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,581 @@
+    public <T extends Filter> T createFilter(Class<T> clazz) throws ServletException {
+        requireNonNull(clazz, "clazz");
+        try {
+            return clazz.getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create a filter: " + clazz.getSimpleName(), e);
+        }
+    }
+
+    @Override
+    public javax.servlet.FilterRegistration getFilterRegistration(String filterName) {
+        requireNonNull(filterName, "filterName");
+        return filterRegistrationMap.get(filterName);
+    }
+
+    @Override
+    public Map<String, FilterRegistration> getFilterRegistrations() {
+        return filterRegistrationMap;
+    }
+
+    @Override
+    public SessionCookieConfig getSessionCookieConfig() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingModes) {
+        ensureUninitialized("setSessionTrackingModes");
+        requireNonNull(sessionTrackingModes, "sessionTrackingModes");
+        sessionTrackingModeSet = sessionTrackingModes;
+    }
+
+    @Override
+    public Set<SessionTrackingMode> getDefaultSessionTrackingModes() {
+        return defaultSessionTrackingModeSet;
+    }
+
+    @Override
+    public Set<SessionTrackingMode> getEffectiveSessionTrackingModes() {
+        return sessionTrackingModeSet;
+    }
+
+    @Override
+    public void addListener(String className) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public <T extends EventListener> void addListener(T listener) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public void addListener(Class<? extends EventListener> listenerClass) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public <T extends EventListener> T createListener(Class<T> clazz) throws ServletException {
+        requireNonNull(clazz, "clazz");
+        try {
+            return clazz.getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create a listener: " + clazz.getSimpleName(), e);
+        }
+    }
+
+    @Override
+    public JspConfigDescriptor getJspConfigDescriptor() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public ClassLoader getClassLoader() {
+        return getClass().getClassLoader();
+    }
+
+    @Override
+    public void declareRoles(String... roleNames) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public String getVirtualServerName() {
+        return ServletUtil.getServerInfo()
+                          .concat(" (localhost")
+                          .concat(":")
+                          .concat(SystemPropertyUtil.get("user.name"))
```

Comment:
We should not just append `null` when `SystemPropertyUtil.get("user.name")` is `null`.